episodes/24-linear-reg-broom.Rmd

```diff
@@ -0,0 +1,323 @@
+
+
+Let's revisit our initial linear model:
+```{r}
+reg_LivEnv_health <- lm(health_london_rank ~ livingEnv_london_rank, data = lon_dims_imd_2019)
+
+summary(reg_LivEnv_health)
+```
+
+There is a lot of useful information, but it not available in a way so that you can combine it with other models or do further analysis.
+We can convert this to tabular data using the 'tidy' function.
+
+```{r}
+tidy(reg_LivEnv_health)
+```
+The row names have been moved into a column called term, and the column names are simple and consistent (and can be accessed using $).
+
+Information about the model can be explored with 'augment'. The function augments the original data with information from the model,
+such as the fitted values and residuals for each of the original points in the regression.
+
+
+```{r}
+augment(reg_LivEnv_health)
+```
+
+Some of the data presented by 'augment' will be discussed in the episode Linear Regression Diagnostics.
+
+Summary statistics are computed for the entire regression, such as R^2 and the F-statistic can be accessed with the 'glance' function:
+
+
+```{r}
+glance(reg_LivEnv_health)
+```
+### Generalised linear models
+
+We can also use the 'broom' functions to present data from Generalised linear and non-linear models.
+For example, if we wanted to explore the Income Rank in relation to whether or not an area was within the City of London.
+
+```{r}
+# add a variable to indicate whether or not an area is within the City of London
+lon_dims_imd_2019 <- lon_dims_imd_2019 %>% mutate(city = la19nm == "City of London")
+
+# create a Generalised Linear Model
+glmlondims <- glm(city ~ Income_london_rank, lon_dims_imd_2019, family = "binomial")
+summary(glmlondims)
+```
+
+Use of 'tidy':
+```{r}
+tidy(glmlondims)
+```
+
+Use of 'augment':
+```{r}
+augment(glmlondims)
+```
+
+Use of 'glance':
+```{r}
+glance(glmlondims)
+```
+You will notice that the statistics computed by 'glance' are different for glm objects than for lm (e.g. deviance rather than R^2).
+
+### Hypothesis Testing
+
+The tidy function can also be applied a range of hypotheses tests, such as built-in functions like t.test, cor.test, and wilcox.test.
+
+### t-test
+```{r}
+tt <- t.test(Income_london_rank ~ city, lon_dims_imd_2019)
+tidy(tt)
+```
+Some cases might have fewer columns (for example, no confidence interval).
+
+Wilcox test:
+```{r}
+wt <- wilcox.test(Income_london_rank ~ city, lon_dims_imd_2019)
+tidy(wt)
+```
+
+Since the 'tidy' output is already only one row, glance returns the same output:
+```{r}
+# t-test
+glance(tt)
+
+# Wilcox test
+glance(wt)
+```
+
+The 'augment' method is defined only for chi-squared tests, since there is no meaningful sense, for other tests,
+in which a hypothesis test produces output about each initial data point.
+
+```{r}
+# xtabs creates a frequency table of IMD deciles within London borooughs
+chit <- chisq.test(xtabs(~ la19nm + IDAOP_london_decile, data = lon_dims_imd_2019))
```

Comment:
This produced the following warning:

```
Warning in chisq.test(xtabs(~la19nm + IDAOP_london_decile, data =
lon_dims_imd_2019)): Chi-squared approximation may be incorrect
```

If that's intentional, we probably want to briefly address this?

Reply:
No, it wasn't. I'll add some text about Chi-Sq assumptions. One of these has probably been violated causing the issue.